src/lib.rs

```diff
@@ -106,11 +119,97 @@ impl SimpleMTKCore for SimpleMTK {
         self.token_id = self.token_id.saturating_add(1);
     }
 
+    /// Burns a token.
+    ///
+    /// Requirements:
+    /// * sender MUST have sufficient amount of token.
+    ///
+    /// Arguments:
+    /// * `id`: Token ID.
+    /// * `amount`: Token's amount to be burnt.
     fn burn(&mut self, id: TokenId, amount: u128) {
         MTKCore::burn(self, vec![id], vec![amount]);
         let sup = self.supply(id);
         let mut _balance = self
             .supply
             .insert(self.token_id, sup.saturating_sub(amount));
     }
+
+    /// Transforms FT tokens to multiple NFTs.
+    ///
+    /// Requirements:
+    /// * a sender MUST have sufficient amount of tokens to burn,
+    /// * a sender MUST be the owner.
+    ///
+    /// Arguments:
+    /// * `id`: Token ID.
+    /// * `amount`: Token's amount to be burnt.
+    /// * `accounts`: To which accounts to mint NFT.
+    /// * `nft_ids`: NFTs' IDs to be minted.
+    /// * `nfts_metadata`: NFT's metadata.
+    fn transform(&mut self, id: TokenId, amount: u128, nfts: Vec<BurnToNFT>) {
+        // pre-checks
+        let mut nft_count = 0;
+        for info in &nfts {
+            nft_count += info.nfts_ids.len();
+        }
+        if amount as usize != nft_count {
+            panic!("MTK: amount of burnt tokens MUST be equal to the amount of nfts.");
+        }
+
+        // burn FT (not to produce another message - just simply use burn_impl)
+        self.assert_can_burn(&msg::source(), &id, amount);
+        self.burn_impl(&id, amount);
+
+        let sup = self.supply(id);
+        let mut _balance = self
+            .supply
+            .insert(self.token_id, sup.saturating_sub(amount));
+        let mut ids = vec![];
+
+        for burn_info in &nfts {
+            if burn_info.account.is_zero() {
+                panic!("MTK: Mint to zero address");
+            }
+            if burn_info.nfts_ids.len() != burn_info.nfts_metadata.len() {
+                panic!("MTK: ids and amounts length mismatch");
+            }
+            burn_info
+                .nfts_metadata
+                .iter()
+                .cloned()
+                .enumerate()
+                .for_each(|(i, meta)| {
+                    self.mint_impl(&burn_info.account, &burn_info.nfts_ids[i], NFT_COUNT, meta)
+                });
+            for id in &burn_info.nfts_ids {
+                ids.push(*id);
+            }
```

Comment:
If you don't use `nfts` later, you can consume it. This eliminates useless cloning and dereferencing.
```suggestion
        for burn_info in nfts {
            if burn_info.account.is_zero() {
                panic!("MTK: Mint to zero address");
            }
            if burn_info.nfts_ids.len() != burn_info.nfts_metadata.len() {
                panic!("MTK: ids and amounts length mismatch");
            }
            burn_info
                .nfts_metadata
                .into_iter()
                .enumerate()
                .for_each(|(i, meta)| {
                    self.mint_impl(&burn_info.account, &burn_info.nfts_ids[i], NFT_COUNT, meta)
                });
            for id in burn_info.nfts_ids {
                ids.push(id);
            }
```